sklearn/linear_model/coordinate_descent.py

```diff
@@ -692,7 +690,7 @@ def fit(self, X, y, check_input=True):
         if self.selection not in ['cyclic', 'random']:
             raise ValueError("selection should be either random or cyclic.")
 
-        if not self.warm_start or self.coef_ is None:
+        if not self.warm_start or getattr(self, "coef_", None) is None:
```

Comment:
I don't think in general we should allow `coef_ = None` to be set before a `warm_start` fit; better that it explicitly be set to zeroes. But there may be user code that does so, copying the dict learning code.